source/utilities/imgcat

```diff
@@ -72,7 +84,7 @@ function print_image() {
 
 function print_payload() {
     if [[ -n "$1" ]]; then
-      printf 'name='`printf "%s" "$1" | base64`";"
+      printf 'name='`printf "%s" "$1" | b64`";"
     fi
 
     VERSION=$(base64 --version 2>&1)
```

Comment:
Could we use the existing test for the base64 version instead of introducing a new one?

Reply:
I thought of doing that, but two things prevented it:

* I don't know what the output-wrapping behavior of `fourmilab` version of `base64` is, so I couldn't address that case

* it's ultimately just the specific behavior of GNU's `-w0` option that I really care about: for all I know, `fourmilab` follows the same behavior and so the direct method of just seeing if `base64` understands `-w0` is more accurate anyway

Reply:
fourmilab always wraps at 72 characters.

Actually, I'm puzzled why this was necessary at all. iTerm2 ignores newlines.

Reply:
Eventually, the output of `base64` ends up being used in `print_payload()`, where `wc -c` counts the characters in the output...so if `base64` produces wrapped output (ie, more newlines), `wc -c` will output the wrong character count.

Reply:
That is true. The size isn't currently used for inline images, but it is used for downloads to show the progress indicator. I would recommend:

* Hoist the version check to the top level and make BASE64ARG a global variable so the test only needs to be done once
* Modify BASE64ARG to include -w0 for GNU only
* Use the b64 function so both the filename and payload get the proper arguments

Reply:
Eh, that's more changes than I'm willing to make. :-) I'll just keep my own copy of `imgcat`. Thanks for looking!

Reply:
I made the change, let me know if it works for you.